### PR TITLE
Add support for journald on Podman

### PR DESCRIPTION
**Pull request by mazzy89**
Signed-off-by: Salvatore Mazzarino <dev@mazzarino.cz>

This should bring the support for journald logs into podman shipped for the UDM PRO

### CODE REVIEW

File: podman-update/build/Dockerfile
Comment (boostchicken) [inline podman-update/build/Dockerfile:5]: Duplicate to new DOckerfile called Dockerfile.Se

File: podman-update/build/podman.Makefile.patch
Comment (boostchicken) [inline podman-update/build/podman.Makefile.patch:10]: Duplicate this into a file called podman.Makefile.se.patch

**Review by boostchicken — CHANGES_REQUESTED** [review]
make new files to not  change current functionalty, ill add new builds.

File: podman-update/build/Dockerfile
Comment (mazzy89) [reply podman-update/build/Dockerfile:5]: gotcha. so the idea would be to create a separate podman binary.

**Comment by mazzy89** [comment]
@boostchicken all the references should be fine now.

Can we run the builds in the branch here before pushing this to master?

**Review by boostchicken — CHANGES_REQUESTED** [review]
You are still using your old patch file in the Dockerfile for the makefiles. make sure to update that

**Comment by mazzy89** [comment]
@boostchicken I've updated it in the last commit. please check it

**Comment by boostchicken** [comment]
yeah its possible, i gotta write you a new build file ill do it here shortly

**Comment by mazzy89** [comment]
Thanks a lot. very appreciated it.

**Comment by boostchicken** [comment]
The next step is pulling those artifcats and making releases out of them, I am happy to grant you permission to do it.

**Comment by mazzy89** [comment]
I think you would need me to run those workflow.

I get 
> First-time contributors need a maintainer to approve running workflows. Learn more.

**Comment by mazzy89** [comment]
> The next step is pulling those artifcats and making releases out of them, I am happy to grant you permission to do it.

Fabolous. Thanks. I would be happy to do that.

**Comment by mazzy89** [comment]
ok the build failed
```
/usr/lib/gcc-cross/aarch64-linux-gnu/6/../../../../aarch64-linux-gnu/bin/ld: cannot find -lsystemd
collect2: error: ld returned 1 exit status
make: *** [bin/conmon] Error 1
```

i will investigate on that.

**Comment by boostchicken** [comment]
Ok you got colllab accessm dont bork existing stuff like you almost did, other than that, go nuts.

**Comment by boostchicken** [comment]
Fyi, if you want systemd for your makefile i dont think stretch will work,  You don't hahve to build off the debian golang images, find a build image tht has systemd support, install go as part off the Dockerfile anad buld away

**Comment by mazzy89** [comment]
Yeah I was thinking the same. I will go with a full OS image w/ systemd support

**Comment by boostchicken** [comment]
or if you wanna use stretch, make sure systemd and its libs and devel are installed 

https://packages.debian.org/stretch/systemd

**Comment by boostchicken** [comment]
GETTTING CLOSE!

**Comment by boostchicken** [comment]
just missing headerfiles now

**Comment by boostchicken** [comment]
apt-get install systemd-dev should do the trick on that missing header file

**Comment by mazzy89** [comment]
yeah let's see if it will work.

**Comment by boostchicken** [comment]
yeah, btw these builds cost me cash, i am not begging for money, if yu can find a way to contribute, please do, if you can't totally understand.

**Comment by boostchicken** [comment]
we have long since smoked the free tier :)

**Comment by mazzy89** [comment]
oh yeah it makes sense. i thought we were in a free tier or something. then i won't mess up with that. i will try another tries and in case it won't work i will move the build process on my servers and once it will be really ready i will make the final commit here. i have an amd64 machine somewhere that i could use to make builds.

**Comment by mazzy89** [comment]
ok installing these dependencies here did not really helped.

Moving these attempts on my servers and once really ready I will make a commit

**Comment by mazzy89** [comment]
@boostchicken we should be ready to got. fingers crossed. tested on my servers and it runs

**Comment by mazzy89** [comment]
now the build is green. time to test.

**Comment by mazzy89** [comment]
the `podman info` output now looks promising

```
host:
  arch: arm64
  buildahVersion: 1.23.1
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - memory
  - devices
  - freezer
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.30, commit: '
  cpus: 4
  distribution:
    codename: stretch
    distribution: debian
    version: "9"
  eventLogger: file
  hostname: UDM-PRO-SE
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.19.152-ui-alpine
  linkmode: dynamic
  logDriver: journald
  memFree: 242028544
  memTotal: 4138082304
  ociRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: |-
      runc version 1.0.2
      commit: v1.0.2-0-g52b36a2d-dirty
      spec: 1.0.2-dev
      go: go1.16.10
      libseccomp: 2.3.1
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_MKNOD,CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 7317221376
  swapTotal: 7516188672
  uptime: 23h 35m 45.41s (Approximately 0.96 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 3
    paused: 0
    running: 3
    stopped: 0
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus: {}
  imageStore:
    number: 3
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.4.2
  Built: 1638541357
  BuiltTime: Fri Dec  3 15:22:37 2021
  GitCommit: 2ad1fd3555de12de34e20898cc2ef901f08fe5ed-dirty
  GoVersion: go1.16.10
  OsArch: linux/arm64
  Version: 3.4.2
```

**Comment by mazzy89** [comment]
going to replace it and migrate the containers to this one. once the final test is done we can merge this.

**Comment by mazzy89** [comment]
@boostchicken i confirm here. the built podman w/ all the dependencies works perfectly and once `log_driver="journald` is set into the `containers.conf` the logs are written in systemd.

Much better way to manage logs for ours UDM PRO SE.

We can merge it.

**Comment by mazzy89** [comment]
Thanks a lot for pushing this out. Please ping me @boostchicken if you need help anyhow with this repo. I will be glad to help in case any SE requests come in.

**Review by boostchicken — APPROVED** [review]

**Comment by boostchicken** [comment]
@mazzy89 you are merged, ill leave you as maintainer of all the UDMSE stuff

**Pull request merged**
